<launch>

<!-- Start image filter node -->
  <node pkg="uv_project" type="image_filter.py" name="image_filter" output="screen"/>

<!-- Launch the depth_image_proc nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

<!-- Start the camerat tf broacaster node -->
  <node pkg="uv_project" type="camera_tf_broadcaster.py" name="camera_tf_broadcaster" output="screen"/>

<!-- Start the transform pointcloud node -->
  <node pkg="uv_project" type="transform_pcl.py" name="transform_pcl" output="screen"/>

<!-- Launch realsense camera rgbd file -->
  <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" >
    <!--
	      LOWEST RESOLUTION, but also has the lowest minimum depth
	      (105mm Min-Z) below which objects generate bad noise, such as
	      when the arm and gripper are raised close to the camera.
    -->    
    <arg name="depth_width"  value="424"/>
    <arg name="depth_height" value="240"/>
    <arg name="color_width"  value="424"/>
    <arg name="color_height" value="240"/> 
    <arg name="depth_fps"    value="30"/>
    <arg name="color_fps"    value="30"/>
    
    </include>

<!-- Launch the depth_image_proc nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="nodelet1"
        args="load depth_image_proc/point_cloud_xyz nodelet_manager">
    <remap from="camera_info"  to="/camera/depth/camera_info"/>
    <remap from="image_rect"   to="/filtered_depth_image"/>
    <remap from="points"       to="/pcl_depthmap"/>
  </node>

 

</launch>
